File: 88.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i = m
        j = n
        while i > 0 and j > 0:
            if nums1[i-1] >= nums2[j-1]:
                nums1[i+j-1] = nums1[i-1]
                i -= 1
            else:
                nums1[i+j-1] = nums2[j-1]
                j -= 1
        if j > 0:
            nums1[:j] = nums2[:j]

File: 88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        while (m >= 0 || n >= 0) {
            if (m == 0) {
                for (int i = 0; i < n; i++) {
                    nums1[i] = nums2[i];
                }
                break;
            }
            if (n == 0) break;
            if (nums1[m-1] < nums2[n-1]) {
                nums1[m+n-1] = nums2[--n];
            } else {
                nums1[m+n-1] = nums1[m-1];
                nums1[m-1] = 0;
                m--;
            }
        }
    }
}